packages/yoroi-extension/app/api/ada/transactions/shelley/transactions.js

```diff
@@ -1084,6 +1088,46 @@ async function newAdaUnsignedTxFromUtxoForConnector(
 
 type UtxoOrAddressing = CardanoAddressedUtxo | {| ...Address, ...Addressing |};
 
+export async function signTransactionFromWallet(
+  senderUtxos: Array<CardanoAddressedUtxo>,
+  unsignedTx: RustModule.WalletV4.Transaction |
+    RustModule.WalletV4.TransactionBuilder |
+    RustModule.WalletV4.TransactionBody |
+    Buffer |
+    Uint8Array,
+  wallet: PublicDeriver<>,
+  password: string,
+  metadata: ?RustModule.WalletV4.AuxiliaryData,
+  otherRequiredSigners: Array<{| ...Address, ...Addressing |}> = [],
+): Promise<RustModule.WalletV4.Transaction> {
+  const withSigning = asGetSigningKey(wallet);
+  if (withSigning == null) {
+    throw new Error('unexpected missing asGetSigningKey result');
+  }
+  const signingKey = await withSigning.getSigningKey();
+  const normalizedKey = await withSigning.normalizeKey({
+    ...signingKey,
+    password,
+  });
+  const withStakingKey = asGetAllAccounting(withSigning);
+  if (withStakingKey == null) {
+    throw new Error('unexpected missing asGetAllAcccounting result');
+  }
+  const stakingKey = await genOwnStakingKey({
+    publicDeriver: withStakingKey,
+    password,
+  });
+  return signTransaction(
+    senderUtxos,
+    unsignedTx,
+    wallet.getParent().getPublicDeriverLevel(),
+    RustModule.WalletV4.Bip32PrivateKey.from_hex(normalizedKey.prvKeyHex),
+    stakingKey,
+    metadata,
+    otherRequiredSigners,
+  )
+}
+
 export function signTransaction(
```

Comment:
This function is no longer used by other modules now that there is `signTransactionFromWallet`  so maybe remove the `export` ?

Reply:
@yushih , it's actually used directly in one place for signing Byron txs, so the same kind of the wallet is not available in that place and calling `signFromWallet` would not work there. The generic signing is compatible and it's just passing null for staking key so that part is ignored.